compiler/luci/pass/src/helpers/LayerInfoMap.cpp

```diff
@@ -22,6 +22,134 @@
+    case luci::CircleOpcode::SPLIT:
+    case luci::CircleOpcode::SPLIT_V:
+    case luci::CircleOpcode::TOPK_V2:
+    case luci::CircleOpcode::UNIQUE:
+    case luci::CircleOpcode::UNPACK:
+    case luci::CircleOpcode::WHILE:
+      return true;
+    // TODO: support this op
+    case luci::CircleOpcode::BIDIRECTIONAL_SEQUENCE_LSTM:
+      throw std::runtime_error("Unsupported op now");
+    default:
+      return false;
+  }
+}
+
+luci::CircleNode *get_multi_output_node(luci::CircleNode *node)
+{
+  switch (node->opcode())
+  {
+    // The following nodes denote outputs of multiple-output nodes.
+    case luci::CircleOpcode::CIRCLECUSTOMOUT:
+    {
+      const auto custom_out = loco::must_cast<CircleCustomOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(custom_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEIFOUT:
+    {
+      const auto if_out = loco::must_cast<CircleIfOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(if_out->input());
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV4OUT:
+    {
+      const auto non_max_sus_v4_out = loco::must_cast<CircleNonMaxSuppressionV4Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v4_out->input());
+    }
+    case luci::CircleOpcode::CIRCLENONMAXSUPPRESSIONV5OUT:
+    {
+      const auto non_max_sus_v5_out = loco::must_cast<CircleNonMaxSuppressionV5Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(non_max_sus_v5_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITOUT:
+    {
+      const auto split_out = loco::must_cast<CircleSplitOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(split_out->input());
+    }
+    case luci::CircleOpcode::CIRCLESPLITVOUT:
+    {
+      const auto splitv_out = loco::must_cast<CircleSplitVOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(splitv_out->input());
+    }
+    case luci::CircleOpcode::CIRCLETOPKV2OUT:
+    {
+      const auto top_kv2_out = loco::must_cast<CircleTopKV2Out *>(node);
+      return loco::must_cast<luci::CircleNode *>(top_kv2_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNIQUEOUT:
+    {
+      const auto unique_out = loco::must_cast<CircleUniqueOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unique_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEUNPACKOUT:
+    {
+      const auto unpack_out = loco::must_cast<CircleUnpackOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(unpack_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEWHILEOUT:
+    {
+      const auto while_out = loco::must_cast<CircleWhileOut *>(node);
+      return loco::must_cast<luci::CircleNode *>(while_out->input());
+    }
+    case luci::CircleOpcode::CIRCLEBIDIRECTIONAL_SEQUENCE_LSTM_OUT:
+      throw std::runtime_error("Unsupported op now");
+    default:
+      return nullptr;
+  }
+}
+
+bool check_layer_info_equal(LayerInfo &left, LayerInfo &right)
```

Comment:
Function name looks a bit ambiguous. How about using the below name?

```suggestion
bool same_setting(LayerInfo &left, LayerInfo &right)
```